Sources/BranchSDK/BNCEncodingUtils.m

```diff
@@ -468,9 +458,12 @@ + (NSData *) dataFromHexString:(NSString*)string {
     }
 
     // If highValue != -1 then we got an odd number of hex values, which is an error.
-    if (highValue == -1)
+    if (highValue == -1) {
         data = [NSData dataWithBytes:bytes length:b-bytes];
+    }
 
+    // Error handling in C code is one case where goto can improve readability.
+    // https://www.kernel.org/doc/html/v4.19/process/coding-style.html
```

Comment:
(Out of scope) goto/label usage aside, the same label name is used throughout with different routines. For readability, the name of the label should be more specific than `exit`.

Reply:
Yea I wanted to do it, but felt like I was expanding scope of the ticket from clarifying logs to clarifying/fixing all kinds of little things. I stopped after adding those comments. Really should come back and make that change.

Reply:
I'll make a follow up for this.